File: VB/WindowsFormsApplication88/CustomColorPickEdit.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Popup

Namespace WindowsFormsApplication88
    <ToolboxItem(True)> _
    Public Class CustomColorPickEdit
        Inherits ColorPickEdit

        Shared Sub New()
            RepositoryItemCustomColorPickEdit.RegisterCustomColorPickEdit()
        End Sub

        Public Sub New()
        End Sub

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Shadows ReadOnly Property Properties() As RepositoryItemCustomColorPickEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomColorPickEdit)
            End Get
        End Property

        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return RepositoryItemCustomColorPickEdit.CustomEditName
            End Get
        End Property

        Protected Overrides Function CreatePopupForm() As PopupBaseForm
            Return New CustomColorPickEditPopupForm(Me)
        End Function
    End Class
End Namespace

File: CS/WindowsFormsApplication88/CustomColorPickEdit.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Popup;

namespace WindowsFormsApplication88 {
    [ToolboxItem(true)]
    public class CustomColorPickEdit : ColorPickEdit {
        static CustomColorPickEdit() {
            RepositoryItemCustomColorPickEdit.RegisterCustomColorPickEdit();
        }

        public CustomColorPickEdit() {
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomColorPickEdit Properties {
            get {
                return base.Properties as RepositoryItemCustomColorPickEdit;
            }
        }

        public override string EditorTypeName {
            get {
                return RepositoryItemCustomColorPickEdit.CustomEditName;
            }
        }

        protected override PopupBaseForm CreatePopupForm() {
            return new CustomColorPickEditPopupForm(this);
        }
    }
}
